src/wp-includes/fonts.php

```diff
@@ -185,8 +158,28 @@ function _wp_filter_font_directory( $font_dir ) {
 	 *     @type string       $baseurl URL path without subdir.
 	 *     @type string|false $error   False or error message.
 	 * }
+	 * @param array $upload_dir The uploads directory data as returned by {@see wp_upload_dir()}.
 	 */
-	return apply_filters( 'font_dir', $font_dir );
+	$font_dir = apply_filters( 'font_dir', $font_dir, $upload_dir );
+
+	// Do not attempt to create the /fonts directory if there was an error in wp_upload_dir() or after the 'font_dir' filter.
+	if ( ! empty( $font_dir['error'] ) ) {
+		return $font_dir;
+	}
+
+	if ( $create_dir ) {
+		$created = wp_mkdir_p( $font_dir['path'] );
+
+		if ( false === $created ) {
+			$font_dir['error'] = sprintf(
+				/* translators: %s: Directory path. */
+				__( 'Unable to create directory %s. Is its parent directory writable by the server?' ),
+				esc_html( $font_dir['path'] )
+			);
+		}
+	}
```

Comment:
This should match the code in `wp_uploads_dir` for creating a directory & caching the results of any failures. 

In https://github.com/WordPress/gutenberg/issues/59387 I was told a single font face can have multiple files attached so this function may be called multiple times.

See https://github.com/WordPress/wordpress-develop/blob/abc261067ac2a878c71118f371432e7bd2c2dc9d/src/wp-includes/functions.php#L2389-L2411

Reply:
Yea, was thinking caching the result may be good here despite that this function will be used a lot less frequently than `wp_upload_dir()`. Will add that.

Reply:
Looked a bit more at how caching works in `wp_upload_dir()`. Using PHP [static variables](https://www.php.net/manual/en/language.variables.scope.php#language.variables.scope.static) is not a good way to do caching. As far as I remember there were some problems when this was added to core, mostly for large sites that have memory caching and file replication. At the same time I think it was not possible to use the standard WP caching mechanisms (however cannot find that discussion now..).

So thinking we can either try to add a "standard" caching using the WP API, or maybe investigate the use cases a bit more and then decide?

Reply:
> So thinking we can either try to add a "standard" caching using the WP API, or maybe investigate the use cases a bit more and then decide?

Yeah, this can probably be done later (possibly with a similar change to the generic upload directory). As font deletion doesn't attempt to create the directory I don't think there is a rush.

Reply:
> I don't think there is a rush

Ok, I'll try to figure out what were the problems with adding this to `wp_upload_dir()` and will take if from there.